api/server/v1/tx.go

```diff
@@ -21,24 +21,70 @@ import (
 )
 
 var (
-	methodPrefix             = "/tigrisdata.v1.Tigris/"
-	insert                   = methodPrefix + "Insert"
-	replace                  = methodPrefix + "Replace"
-	update                   = methodPrefix + "Update"
-	delete                   = methodPrefix + "Delete"
-	read                     = methodPrefix + "Read"
-	commitTransaction        = methodPrefix + "CommitTransaction"
-	rollbackTransaction      = methodPrefix + "RollbackTransaction"
-	dropCollection           = methodPrefix + "DropCollection"
-	listCollection           = methodPrefix + "ListCollections"
+	methodPrefix = "/tigrisdata.v1.Tigris/"
+
+	insert  = methodPrefix + "Insert"
+	replace = methodPrefix + "Replace"
+	update  = methodPrefix + "Update"
+	deleteR = methodPrefix + "Delete"
+	read    = methodPrefix + "Read"
+
+	search = methodPrefix + "Search"
+
+	//publish   = methodPrefix + "Publish"
+	subscribe = methodPrefix + "Subscribe"
+
+	events = methodPrefix + "Events"
+
+	//beginTransaction    = methodPrefix + "BeginTransaction"
+	commitTransaction   = methodPrefix + "CommitTransaction"
+	rollbackTransaction = methodPrefix + "RollbackTransaction"
+
 	createOrUpdateCollection = methodPrefix + "CreateOrUpdateCollection"
+	dropCollection           = methodPrefix + "DropCollection"
+
+	//createDatabase = methodPrefix + "CreateDatabase"
+	//dropDatabase   = methodPrefix + "DropDatabase"
+
+	listDatabases   = methodPrefix + "ListDatabases"
+	listCollections = methodPrefix + "ListCollections"
+
+	describeDatabase   = methodPrefix + "DescribeDatabase"
+	describeCollection = methodPrefix + "DescribeCollection"
 )
 
+func isRead(name string) bool {
+	switch name {
+	case read, events, search, subscribe:
+		return true
+	case listCollections, listDatabases:
+		return true
+	case describeCollection, describeDatabase:
+		return true
+	default:
+		return false
+	}
+}
+
+func isWrite(name string) bool {
+	return !isRead(name)
```

Comment:
we classified requests in three categories for metrics query api. read, write and metadata. https://github.com/tigrisdata/tigris/blob/main/server/services/v1/observability.go#L213 from quota perspective what you did makes sense but just bringing it up here if we should restrict the usage of N number of db creation for example.

Reply:
Yeah we don't count list database, list collection, etc as read queries when displaying in the web console

Reply:
Right, there is discrepancy, but we need to enforce quota even for metadata requests, so we classify them as reads and writes here.

Reply:
Fixed. There are now only two categories for the quota metrics request. Check: `FormDatadogQueryNoMeta`